engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java

```diff
@@ -993,16 +998,75 @@ public Pair<NicProfile, Integer> allocateNic(final NicProfile requested, final N
         }
 
         deviceId = applyProfileToNic(vo, profile, deviceId);
-
+        if (vm.getType() == Type.DomainRouter) {
```

Comment:
Isn't it better to get the mtu value here and use it here and in ln 1009-1011 instead of querying router twice?